lib/galaxy/datatypes/phylip.py

```diff
@@ -59,12 +59,22 @@ def sniff_prefix(self, file_prefix):
         f = file_prefix.string_io()
         # Get number of sequence from first line
         nb_seq = int(f.readline().split()[0])
-        assert nb_seq > 0
+        if nb_seq <= 0:
+            return False
         # counts number of sequence from first stack
         count = 0
         for line in f:
             if not line.split():
                 break
+            if len(line) < 11:
+                # Sequence characters immediately follow the sequence ID.
+                # They must start at the 11th character in the line, as the first 10 characters are reserved for the sequence ID
+                return False
+            # Fail if sequence is not ascii
+            line[10:].decode('ascii')
```

Comment:
```suggestion
            line[10:].encode('ascii')
```

Reply:
I can't believe I can still make this mistake after all the python 3 porting we've done ...